features/kafka/producer/src/main/java/org/opennms/features/kafka/producer/OpennmsKafkaProducer.java

```diff
@@ -168,6 +199,103 @@ public void destroy() {
+            LOG.error("forwardTopologyMessage: null protocol");
+            return;
+        }
+        if (message.getMessagestatus() == null) {
+            LOG.error("forwardTopologyMessage: null status");
+            return;
+        }
+        if (message.getMessagebody() == null ) {
+            LOG.error("forwardTopologyMessage: null message");
+            return;            
+        }
+
+        if (message.getMessagestatus() == TopologyMessageStatus.DELETE) {
+            forwardTopologyDeleteMessage(message);
+        } else {
+            forwardTopologyUpdateMessage(message);
+        }            
+    }
+
+    private void forwardTopologyDeleteMessage(OnmsTopologyMessage message) {
+        OpennmsModelProtos.TopologyRef mappedTopomsg = 
+                protobufMapper.toTopologyRef(message.getProtocol().getId(), message.getMessagebody().getId()).build();
+        if (message.getMessagebody() instanceof OnmsTopologyVertex) {
+            forwardTopologyVertexMessage(mappedTopomsg.toByteArray(),null);
+        } else if (message.getMessagebody() instanceof OnmsTopologyEdge) {
+            forwardTopologyEdgeMessage(mappedTopomsg.toByteArray(),null);
+        } else if (message.getMessagebody() instanceof OnmsTopologySegment) {
+            forwardTopologyVertexMessage(mappedTopomsg.toByteArray(),null);
+        } else {
+            LOG.error("forwardTopologyDeleteMessage: no supported update class {}", message.getMessagebody().getClass().getName());            
+        }
+    }
+
+    private void forwardTopologyUpdateMessage(OnmsTopologyMessage message) {
+        if (message.getMessagebody() instanceof OnmsTopologyVertex) {
+            OnmsTopologyVertex vertex = (OnmsTopologyVertex) message.getMessagebody();
+            final OpennmsModelProtos.TopologyVertex mappedTopoMsg = 
+                    protobufMapper.toVertexTopologyMessage(message.getProtocol().getId(),vertex).build();
+            forwardTopologyVertexMessage(mappedTopoMsg.getRef().toByteArray(),
+                                         mappedTopoMsg.toByteArray());
+        } else if (message.getMessagebody() instanceof OnmsTopologyEdge) {
+            OnmsTopologyEdge edge = (OnmsTopologyEdge) message.getMessagebody();
+            final OpennmsModelProtos.TopologyEdge mappedTopoMsg = 
+                    protobufMapper.toEdgeTopologyMessage(message.getProtocol().getId(),edge).build();
+            forwardTopologyEdgeMessage(mappedTopoMsg.getRef().toByteArray(),
+                                       mappedTopoMsg.toByteArray());
+        } else if (message.getMessagebody() instanceof OnmsTopologySegment) {
+            OnmsTopologySegment edge = (OnmsTopologySegment) message.getMessagebody();
+            final OpennmsModelProtos.TopologySegment mappedTopoMsg = 
+                    protobufMapper.toSegmentTopologyMessage(message.getProtocol().getId(),edge).build();
+            forwardTopologySegmentMessage(mappedTopoMsg.getRef().toByteArray(),
```

Comment:
Can we model the segments as vertices and send these to the existing "vertex" topic instead of creating a separate topic for the segments? The segments are nothing but "special" vertices.

Reply:
So if I understand is very similar in the way we layout maps....

so then it we should transform the segment into...
Consider we have Segment with 4 ports connectin V1,V2,V3,V4
then this can be translated into

5 Vertex S, V1,V2,V3,V4 and 4 edges
V1-S
V2-S
V3-S
V4-S

Please confirm

Reply:
Correct, if `S` is a shared segment and `V1`, `V2`, `V3`, `V4` are ports attached to this shared segment. Then we can model this as 5 vertices, one for each port and one for the segment, as well as 4 edges connecting the port to the segment.

Reply:
Another solution is to avoid using ege an using only segments....
at the en an ege has only two vertcies and a segment has many.....